### DISCUSSION TITLE
Navigation section collapse

### DISCUSSION THREAD

**Discussion by Pyjong** [squidfunk/mkdocs-material#2173] [General]
Hello,

First of all, MkDocs-Material is great! Finally documentation as I like it. Thanks!

Now to my question. I'm already happy as it is, but I guess it would be even nicer if the navigation wouldn't collapse after I click something in it. I know there is nav.expand but if it was possible to only keep those sections expanded that lead to currently viewed document that would be awesome. Is there perhaps a way to do that?

Thanks.

**Comment by squidfunk** [comment]
__The navigation will always expand the path to the active page and collapse all non-active pages__, except when using the `navigation.expand` feature flag, which will always expand all navigation paths, as you already noted.

This is also true for [instant loading](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#instant-loading), as the whole navigation is always replaced. Material for MkDocs isn't a single-page application, so the state is generally lost/replaced. I agree that it would be ideal to keep the expanded/collapsed states, but it's not trivial to implement. You might try the MkDocs [Windmill](https://gristlabs.github.io/mkdocs-windmill/) theme, which exhibits the desired behavior.

**Reply by Pyjong** [reply]
I see, thanks for the info!

**Comment by jcalabreseSH** [comment]
I saw the nav.expand options will expand the whole nav or we can chose to collapse However, I see only nested menu items collapse. Is there a way to get the main nav items to be collapsed.
For example: in this photo i would like to collapse Developer Portal and Cloud API

<img width="254" alt="Screen Shot 2022-07-29 at 7 51 23 AM" src="https://user-images.githubusercontent.com/98052391/181797248-452f7dc3-fb0c-4a95-ba86-04a0db27ddfa.png">

**Reply by squidfunk** [reply]
Yes, remove the `navigation.sections` feature flag.

**Reply by jasonatcb** [reply]
Thanks, this solve my exact problem.
